implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -243,8 +243,8 @@ protected DistributionStatisticConfig defaultHistogramConfig() {
     @Override
     public void close() {
         stop();
-        if (!isPublishing() && isDelta()) {
-            if (!isDataPublishedForCurrentStep()) {
+        if (config.enabled() && isDelta() && !isClosed()) {
```

Comment:
These aren't directly related to this change but were out-of-sync with logic in StepMeterRegistry which also applies to OTLP.

Reply:
This is going to be problem given this is a duplicate code being copied around.

Reply:
I agree, which is why I wanted to avoid it in the first place, but unfortunately things aren't so easy. We'll have to look again at what we can do to remove the need for the duplication without opening up too much, but that's definitely out of scope for this PR.